include/queue/queue_sycl.hpp

```diff
@@ -0,0 +1,137 @@
+#ifndef QUEUE_SYCL_HPP
+#define QUEUE_SYCL_HPP
+
+#include <CL/sycl.hpp>
+#include <queue/pointer_mapper.hpp>
+#include <queue/queue_base.hpp>
+#include <stdexcept>
+namespace blas {
+
+template <>
+class Queue_Interface<SYCL> {
+  /*!
+   * @brief SYCL queue for execution of trees.
+   */
+  cl::sycl::queue q_;
+  mutable cl::sycl::codeplay::PointerMapper pointer_mapper;
+  bool pointer_mapper_owner;
+  using generic_buffer_data_type = cl::sycl::codeplay::buffer_data_type_t;
+
+ public:
+  enum device_type { UNSUPPORTED_DEVICE, INTELGPU, AMDGPU };
+
+  explicit Queue_Interface(cl::sycl::queue q)
+      : q_(q), pointer_mapper_owner(true) {}
+  // FIXME : The assignment constructor for pointer mapper is deleted
+  /*  Queue_Interface(cl::sycl::queue q,
+                    cl::sycl::codeplay::PointerMapper& pointer_mapper_)
+        : q_(q), pointer_mapper(pointer_mapper_), pointer_mapper_owner(false)
+     {}*/
```

Comment:
create github issue

Reply:
created and the commented code has been removed